Upgrade to Unicode 16.0

wezm opened the pull request.

wezm committed:
Upgrade to Unicode 16.0

adrianwong reviewed (approved):
Looks like the new scripts need to be mapped to their Harfbuzz equivalents. Looks fine otherwise.

wezm committed:
ci: bump actions/checkout

wezm committed:
Drop Harfbuzz feature

We aren't using it and there are no dependants of this crate listed on
crates.io and not results from a search:
`language:toml yeslogic-unicode-script` on GitHub. It complicates
updates by the dependencies being incompatible with older versions
of rustc.

wezm commented:
> Looks like the new scripts need to be mapped to their Harfbuzz equivalents. Looks fine otherwise.

I dropped the Harfbuzz feature:

> We aren't using it and there are no dependants of this crate listed on crates.io and no results from a search: `language:toml yeslogic-unicode-script` on GitHub. It complicates updates by the dependencies being incompatible with older versions of rustc.
